src/attribute-filters.tsx

```diff
@@ -0,0 +1,376 @@
+      filters.severity = new Set(severityValues);
+    } else if (pipelineStage.operator === '|=' && !filters.content) {
+      filters.content = new Set([removeQuotes(pipelineStage.value)]);
+    }
+  }
+
+  return filters;
+};
+
+export const getNamespaceMatcher = (
+  namespace?: string,
+): LabelMatcher | undefined => {
+  if (namespace === undefined) {
+    return undefined;
+  }
+
+  return {
+    label: 'kubernetes_namespace_name',
+    operator: '=',
+    value: `"${namespace}"`,
+  };
+};
+
+export const queryWithNamespace = ({
+  query,
+  namespace,
+}: {
+  query: string;
+  namespace?: string;
+}) => {
+  const logQLQuery = new LogQLQuery(query ?? '');
+
+  logQLQuery.addSelectorMatcher(getNamespaceMatcher(namespace));
+
+  return logQLQuery.toString();
+};
+
+export const getMatcherFromSet = (
+  label: string,
+  values: Set<string>,
+): LabelMatcher | undefined => {
+  const valuesArray = Array.from(values);
+  if (valuesArray.length === 0) {
+    return undefined;
+  }
+
+  if (valuesArray.length === 1) {
+    return {
+      label,
+      operator: '=',
+      value: `"${valuesArray[0]}"` ?? '',
+    };
+  }
+
+  return {
+    label,
+    operator: '=~',
+    value: `"${valuesArray.filter(notEmptyString).join('|')}"`,
+  };
+};
+
+export const getMatchersFromFilters = (
+  filters?: Filters,
+): Array<LabelMatcher> => {
+  if (!filters) {
+    return [];
+  }
+
+  const matchers: Array<LabelMatcher | undefined> = [];
+
+  for (const key of Object.keys(filters)) {
+    const value = filters[key];
+    if (value) {
+      switch (key) {
+        case 'namespace':
+          matchers.push(getMatcherFromSet('kubernetes_namespace_name', value));
+          break;
+        case 'pod':
+          matchers.push(getMatcherFromSet('kubernetes_pod_name', value));
+          break;
+        case 'container':
+          matchers.push(getMatcherFromSet('kubernetes_container_name', value));
+          break;
+      }
+    }
+  }
+
+  return matchers.filter(notUndefined);
+};
+
+export const getContentPipelineStage = (
+  filters?: Filters,
+): PipelineStage | undefined => {
+  if (!filters) {
+    return undefined;
+  }
+
+  const content = filters.content;
+
+  if (!content) {
+    return undefined;
+  }
+
+  const [textValue] = content;
+
+  if (textValue === undefined) {
+    return undefined;
+  }
+
+  return { operator: '|=', value: `"${textValue}"` };
+};
+
+export const getSeverityFilterPipelineStage = (
+  filters?: Filters,
+): PipelineStage | undefined => {
+  if (!filters) {
+    return undefined;
+  }
+
+  const severity = filters.severity;
+
+  if (!severity) {
+    return undefined;
+  }
+
+  const unknownFilter = severity.has('unknown')
+    ? 'level="unknown" or level=""'
+    : '';
+
+  const severityFilters = Array.from(severity).flatMap(
+    (group: string | undefined) => {
+      if (group !== 'unknown' && group !== undefined) {
+        return [severityAbbreviations[group as Severity].join('|')];
```

Comment:
```suggestion
        return severityAbbreviations[group as Severity];
```
Because we are doing the `join('|')` below as well.